accepted/future-releases/class-modifiers/feature-specification.md

```diff
@@ -836,38 +785,82 @@ the latter, even if the class is being used as a mixin in a post-feature
 library where it does happen to be possible to distinguish those two
 intents.*
 
+### Enum classes
+
+The class introduced by an `enum` declaration is considered `final` for any
+purpose where a class modifier is required.
+
+The behavior of `enum` declarations is unchanged. Since an `enum` class cannot
+have any subclasses, the implicit `final` modifier would not prevent any
+otherwise allowed operation.
+The implicit `final` also does not imply fewer restrictions than `enum`
+declarations would otherwise have. The modifier is applied only to automatically
+satisfy any requirements introduced by super-interfaces.
+
 ### Anonymous mixin applications
 
-An *anonymous mixin application* class is a class resulting from a mixin application
-that does not have its own declaration.
+An *anonymous mixin application* class is a class resulting from
+a mixin application that does not have its own declaration.
 That is all mixin applications classes other than the final class
-of a <Code>class C = S with M1, …, M<sub>n</sub>;</code> declaration, the mixin application of <code>M<sub>n</sub></code> to
-the superclass <code>S with M1, …, M<sub>n-1</sub></code>, which is denoted by declaration and name `C`.
+of a <code>class C = S with M1, …, M<sub>n</sub>;</code> declaration,
+the mixin application of <code>M<sub>n</sub></code> to
+the superclass <code>S with M1, …, M<sub>n-1</sub></code>,
+which is denoted by declaration and name `C`.
 
 An anonymous mixin application class cannot be referenced anywhere except in
-the context where the application occurs, so its only role is to be a superclass of
-another class in the same library.
+the context where the application occurs, so its only role is
+to be a superclass of another class in the same library.
+
+To ensure reasonable and correct behavior, without having to special-case such
+anonymous mixin application classes elsewhere, we infer class modifiers as
+follows.
+
+First we define the following transitive property on declarations:
+
+* A declaration *prohibits inheritance* _(of its implementation)_ if:
```

Comment:
Does 'transitive' mean that it is possible for an anonymous mixin application to prohibit inheritance?

In that case, an anonymous mixin application is considered to be a 'declaration', and it 'is marked' with the implicitly induced modifier.

Otherwise, it's impossible for *C* in line 834 to 'prohibit inheritance'.